google/cloud/storage/blob.py

```diff
@@ -3253,7 +3253,10 @@ def _raise_from_invalid_response(error):
              to the failed status code
     """
     response = error.response
-    error_message = str(error)
+    if response.text:
+        error_message = response.text + ": " + str(error)
+    else:
+        error_message = "unknown error: " + str(error)
```

Comment:
I'd prefer not to add the prefix here.  @frankyn WDYT?

Reply:
Could you clarify the difference between response.text and error (add comments I suspect others will ask this question when reviewing the code in the future). 

I also think repsonse.text could be added to the line below when constructing the `message` value instead of this conditional.

Reply:
@frankyn  I have added condition here to match the method which is responsible for other http responses and errors declared in `pytho-api-core.exceptions.from_http_response`  you can find here https://github.com/googleapis/python-api-core/blob/622931721ce34839d630aa1e974c7d8f47b5d25e/google/api_core/exceptions.py#L390-L411  

`response.text: ` gives the actual reason of error
`str(error): ` gives ('Request failed with status code', 404(Error Code), 'Expected one of', <HTTPStatus.OK: 200>, <HTTPStatus.PARTIAL_CONTENT: 206>)

I think here we can remove `str(error)` as it's not meaning full for the users, also not included in other http methods from python-api-core as mention above.

Reply:
@frankyn  I think it's necessary to add actual reason in error message, Ex: in the https://github.com/googleapis/python-storage/issues/249  issue user share a stack trace of error but it's hard to debug as actual reason isn't there in message.

Reply:
I think I need to pull in @andrewsg on this one when he has a moment.

Andrew do you have guidance on raising the actual error that occurred in this case?

Reply:
If I understand correctly, the problem here is the same as I've observed elsewhere where the API responds with human-readable text which is not converted into a specific error message by our client library. Prepending this human-readable text does make sense though I'm neutral on whether we should include the "unknown error" prepending in the `else` case.

We should make sure that the traceback still includes the original function call that caused the error.

Reply:
Thanks @andrewsg.

I don't see the value of `unknown error:` and could be confusing otherwise. Please only preppend `response.text` when it's available. Remove `unknown error:`.